RELEASE/scripts/autoscend/iotms/mr2016.ash

```diff
@@ -1007,181 +1007,51 @@ boolean LX_ghostBusting()
-			}
-		}
-		if(goal == $location[Lair of the Ninja Snowmen])
-		{
-			if(L8_trapperNinjaLair())
-			{
-				return true;
-			}
-		}
-
-		if((goal == $location[The Red Zeppelin]) && (internalQuestStatus("questL11Ron") < 3))
-		{
-			return false;
-		}
-		if (goal == $location[The Hidden Park] && internalQuestStatus("questL11Worship") > 3)
-		{
-			return false;
-		}
-
-		item replaceAcc3 = $item[none];
-		if(goal == $location[Inside The Palindome])
-		{
-			if(!possessEquipment($item[Talisman O\' Namsilat]))
-			{
-				return false;
-			}
-			if(equipped_item($slot[acc3]) != $item[Talisman O\' Namsilat])
-			{
-				replaceAcc3 = equipped_item($slot[acc3]);
-				autoEquip($slot[acc3], $item[Talisman O\' Namsilat]);
-			}
-		}
-
-		if(newbieFail)
-		{
-			auto_log_critical("Can't bust that ghost, we don't feel good!! skipping this ghost...", "red");
-			set_property("questPAGhost", "unstarted");
-			set_property("ghostLocation", "");
-			return false;
-		}
-
-		if((equipped_item($slot[Back]) != $item[Protonic Accelerator Pack]) && can_equip($item[Protonic Accelerator Pack]))
-		{
-			autoEquip($slot[Back], $item[Protonic Accelerator Pack]);
-		}
-
-		auto_log_info("Time to bust some ghosts!!!", "green");
-		boolean advVal = autoAdv(goal);
-		if(replaceAcc3 != $item[none])
-		{
-			autoEquip($slot[acc3], replaceAcc3);
-		}
-		if(have_effect($effect[Beaten Up]) == 0)
-		{
-			set_property("questPAGhost", "unstarted");
-			set_property("ghostLocation", "");
-		}
+		return false;
+	}
+	if(goal == $location[Inside The Palindome] && !possessEquipment($item[Talisman O\' Namsilat]))
+	{
+		return false;
+	}
+	
+	//zone unlocks which require no adv spent. no ghost will not show up here unless zone is available. thus no need to include skip code.
+	startHippyBoatmanSubQuest();	//unlocks $location[The Old Landfill].
+	
+	//zone unlocks which require no adv spent. where a ghost can show up even if you did not unlock the zone. if failed to unlock we skip this ghost
+	startMeatsmithSubQuest();		//unlocks $location[The Skeleton Store]
+	startArmorySubQuest();			//unlocks $location[Madness Bakery]
+	startGalaktikSubQuest();		//unlocks $location[The Overgrown Lot]
+	if($locations[The Skeleton Store, Madness Bakery, The Overgrown Lot] contains goal && !zone_available(goal))
+	{
+		auto_log_critical("Failed to unlock the location [" +goal+ "]. skipping this ghost...", "red");
+		set_property("questPAGhost", "unstarted");
+		set_property("ghostLocation", "");
+		return false;
+	}
 
-		return advVal;
+	auto_log_info("Ghost busting time! At: " +goal, "blue");
+	autoForceEquip($item[Protonic Accelerator Pack]);
+	if(goal == $location[Inside The Palindome])
+	{
+		autoForceEquip($slot[acc3], $item[Talisman O\' Namsilat]);
 	}
-	return false;
+	acquireHP();
+	auto_log_info("Time to bust some ghosts!!!", "green");
+	boolean advVal = autoAdv(goal);
+	if(have_effect($effect[Beaten Up]) == 0)	//is this kind of check still required or does mafia track defeating ghosts now?
```

Comment:
You're correct, this is unnecessary now. Mafia will set `questPAGhost` to `unstarted` once the ghost has been killed and clear `ghostLocation`

```
[541236] The Haunted Conservatory
Preference lastEncounter changed from drunk pygmy to The ghost of Lord Montague Spookyraven
Encounter: The ghost of Lord Montague Spookyraven
Round 0: Malibu Stacey wins initiative!
Round 1: You lose 1 hit point
...
Round 7: Malibu Stacey casts TRAP GHOST!
Round 8: The ghost of Lord Montague Spookyraven takes 1 damage.
Round 8: Malibu Stacey wins the fight!
You acquire an item: psychokinetic energy blob (3)
You acquire an item: Spookyraven signet
After Battle: Bustin' that ghost made you feel really good! (+20 Stats)
After Battle: You gain 17 Strongness
After Battle: You gain 23 Wizardliness
After Battle: You gain 41 Sarcasm
Preference feelNostalgicMonster changed from drunk pygmy to The ghost of Lord Montague Spookyraven
This combat did not cost a turn
Preference questPAGhost changed from started to unstarted
Preference ghostLocation changed from The Haunted Conservatory to 
```

However it's fine to leave in if you want to merge this.

Reply:
thank you. I went ahead and removed that bit of code. will perform further testing now.